File: src/main/java/com/company/payroll/model/Promotion.java
package com.company.payroll.model;

import java.math.BigDecimal;
import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
public class Promotion {
    private Integer pId;

    private BigDecimal currentsalary;

    private BigDecimal promotesalary;

    private LocalDate promotedate;

    private Integer titleno;

    private String comment;

    private Integer eId;

    private Integer mId;
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/model/Promotion.kt
package com.company.payroll.model

import lombok.AllArgsConstructor
import lombok.RequiredArgsConstructor
import java.math.BigDecimal
import java.time.LocalDate

@AllArgsConstructor
@RequiredArgsConstructor
data class Promotion(
    var pId: Int,
    var currentSalary: BigDecimal,
    var promoteSalary: BigDecimal,
    var promoteDate: LocalDate,
    var title: Title,
    var comment: String,
    var staffId: Int
)
